File: J3.L.P0001/src/java/model/Quiz.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package model;

import java.util.Date;
import java.util.List;

/**
 *
 * @author nguyenvanan
 */
public class Quiz {

    private int id;
    private int quantity;
    private List<QA> qa;
    private Date dateStarted;
    private User user;

    public Quiz() {
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getQuantity() {
        return quantity;
    }

    public void setQuantity(int quantity) {
        this.quantity = quantity;
    }

    public List<QA> getQa() {
        return qa;
    }

    public void setQa(List<QA> qa) {
        this.qa = qa;
    }

    public Date getDateStarted() {
        return dateStarted;
    }

    public void setDateStarted(Date dateStarted) {
        this.dateStarted = dateStarted;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

}

File: J3.L.P0001/web/static/script/quiz.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

window.addEventListener("load", function(e){
    var stopDate = new Date("${quiz.dateStop}").getTime();
    var countdown = setInterval(function () {
        var now = new Date().getTime();

        // calc distance between now and count down date
        var timeLeft = stopDate - now;

        // specify days, hours, minutes, and seconds by time left
        var oneSec = 1000;
        var oneMin = 60 * oneSec;
        var oneHour = 60 * oneMin;
        var oneDay = 24 * oneHour;
        var days = Math.floor(timeLeft / oneDay);
        var hours = Math.floor((timeLeft % oneDay) / oneHour);
        var minutes = Math.floor((timeLeft % oneHour) / oneMin);
        var seconds = Math.floor((timeLeft % oneMin) / oneSec);

        var clock = "";

        // format date
        if (days > 0) {
            clock += days + ":";
        }
        if (hours > 0) {
            clock += hours + ":";
        }
        if (minutes > 0) {
            clock += minutes + ":";
        }
        if (seconds > 0) {
            clock += seconds;
        }

        // set to timer
        var timer = document.getElementById("timer");
        timer.innerHTML = clock;

        if (timeLeft <= 0) {
            clearInterval(countdown);
            // auto submit
            document.getElementById("submit").click();
        }

    }, 1000);
});